pgmpy/prediction/NaivelVRegressor.py

```diff
@@ -0,0 +1,163 @@
+        tags = super().__sklearn_tags__()
+        tags.regressor_tags.poor_score = True
+        return tags
+
+    def _prepare_feature_df(self, X) -> pd.DataFrame:
+        """
+        Accept a numpy/pandas dataframe and returns pandas df
+        If numpy array is passed, it converts to pandas df
+        """
+        # Step 1: Get required feature columns
+        required_features = self.feature_columns_
+
+        # Step 2: Convert input to DataFrame format
+        if isinstance(X, pd.DataFrame):
+            X_df = X
+
+        else:
+            # For numpy arrays, use range index as column names
+            X_arr = np.asarray(X)
+            if X_arr.ndim == 1:
+                raise ValueError(
+                    "Reshape your data: X must be 2D. If using a 1D array, reshape it to (n_samples, 1)."
+                )
+            X_df = pd.DataFrame(X_arr, columns=range(X_arr.shape[1]))
+
+        # Step 3: Validation: column names must exactly match DAG variables
+        missing_columns = set(required_features) - set(X_df.columns)
+        if missing_columns:
+            raise ValueError(
+                f"Missing required columns in input data: {list(missing_columns)}. "
+                f"DAG expects columns: {required_features}, but got: {list(X_df.columns)}"
+            )
+
+        return X_df[required_features]
+
+    def fit(self, X, y, sample_weight: Optional[Any] = None):
+        """
+        This method performs two-stage least squares regression using the specified causal graph.
+        It first fits the stage 1 estimator to predict the exposure variable from the instrument
+        and pretreatment variables, then fits the stage 2 estimator to predict the outcome
+        variable from the predicted exposure and pretreatment variables.
+        """
+        validate_data(self, X, y, accept_sparse=False, ensure_2d=True, dtype="numeric")
+        stage1_estimator = clone(self.stage1_estimator)
+        stage2_estimator = clone(self.stage2_estimator)
+
+        exposure_vars = list(self.causal_graph.get_role("exposure"))
+        outcome_vars = list(self.causal_graph.get_role("outcome"))
+        instrument_vars = list(self.causal_graph.get_role("instrument"))
+
+        if len(exposure_vars) != 1:
+            raise ValueError(
+                f"NaiveIVRegressor requires exactly one exposure; got {len(exposure_vars)}"
+            )
+        if len(outcome_vars) != 1:
+            raise ValueError(
+                f"NaiveIVRegressor requires exactly one outcome; got {len(outcome_vars)}"
+            )
+        if len(instrument_vars) < 1:
+            raise ValueError(
+                f"NaiveIVRegressor requires at least one instrument; got {len(instrument_vars)}"
+            )
+
+        self.exposure_var_ = exposure_vars[0]
+        self.outcome_var_ = outcome_vars[0]
+        self.instrument_vars_ = instrument_vars
+        self.pretreatment_vars_ = list(self.causal_graph.get_role("pretreatment"))
```

Comment:
Why do we need to do `list(self.causal_graph.get_role("pretreatment"))`? Doesn't `get_role` always return a list?